scripts/js/charts.js

```diff
@@ -76,61 +76,62 @@ const htmlLegendPlugin = {
 
       boxSpan.addEventListener("click", () => {
         const { type } = chart.config;
+        const isPieOrDoughnut = type === "pie" || type === "doughnut";
 
-        if (type === "pie" || type === "doughnut") {
+        if (isPieOrDoughnut) {
           // Pie and doughnut charts only have a single dataset and visibility is per item
           chart.toggleDataVisibility(item.index);
         } else {
-          chart.setDatasetVisibility(item.datasetIndex, !chart.isDatasetVisible(item.datasetIndex));
+          const isVisible = chart.isDatasetVisible(item.datasetIndex);
+          chart.setDatasetVisibility(item.datasetIndex, !isVisible);
         }
 
         chart.update();
       });
 
       const textLink = document.createElement("p");
-      if (
-        chart.canvas.id === "queryTypePieChart" ||
-        chart.canvas.id === "forwardDestinationPieChart"
-      ) {
+      const isQueryTypeChart = chart.canvas.id === "queryTypePieChart";
+      const isForwardDestinationChart = chart.canvas.id === "forwardDestinationPieChart";
+
+      if (isQueryTypeChart || isForwardDestinationChart) {
         // Text (link to query log page)
-        textLink.title = "List " + item.text + " queries";
+        textLink.title = `List ${item.text} queries`;
 
         textLink.addEventListener("click", () => {
-          if (chart.canvas.id === "queryTypePieChart") {
-            globalThis.location.href = "queries.lp?type=" + item.text;
-          } else if (chart.canvas.id === "forwardDestinationPieChart") {
+          if (isQueryTypeChart) {
+            globalThis.location.href = `queries.lp?type=${item.text}`;
+          } else if (isForwardDestinationChart) {
             // Encode the forward destination as it may contain an "#" character
             const upstream = encodeURIComponent(upstreams[item.text]);
-            globalThis.location.href = "queries.lp?upstream=" + upstream;
+            globalThis.location.href = `queries.lp?upstream=${upstream}`;
           }
         });
       }
 
-      textLink.style.margin = 0;
-      textLink.style.padding = 0;
       textLink.style.textDecoration = item.hidden ? "line-through" : "";
       textLink.className = "legend-label-text";
       textLink.append(item.text);
 
       li.append(boxSpan, textLink);
       ul.append(li);
-    });
+    }
   },
 };
 
 // eslint-disable-next-line no-unused-vars
-var customTooltips = function (context) {
-  var tooltip = context.tooltip;
-  var tooltipEl = document.getElementById(this.chart.canvas.id + "-customTooltip");
+const customTooltips = context => {
+  const tooltip = context.tooltip;
+  const canvasId = context.chart.canvas.id;
```

Comment:
context provides a `chart` object where we can get this info without relying on the "cryptic" `this` value.